docs/source/services.md

```diff
@@ -0,0 +1,84 @@
+# JupyterHub services
+
+JupyterHub 0.7 adds the notion of Services.
+A Service is a process that interacts with the Hub REST API.
+Services may perform actions such as shutting down user servers that have been idle for some time,
+or registering additional web servers that should also use the Hub's authentication
+and be served behind the Hub's proxy.
+
+There are two main characteristics of services:
+
+1. Is it **managed** by JupyterHub?
+2. Does it have a web server that should be added to the proxy?
+
+If a `command` is specified for launching the service, it will be started and managed by the Hub.
+If a `url` is specified for where the service runs its own webserver,
+it will be added to the Hub's proxy at `/service/:service-name`.
+
+## Managed services
+
+**Managed** services are services that the Hub starts and is responsible for.
+These can only be local subprocesses of the Hub.
+While there are similarities with notebook Spawners,
+there are no plans to support the same spawning abstractions as notebook
+If you want to run these services in docker or other environments,
```

Comment:
Would a generic extension mechanism be sufficient? For the most part, I would like to discourage people adding code that runs in the Hub process itself, but that doesn't mean I need to make it impossible.

I could start simple with two hooks:

1. setup, fires after all Hub initialization, before starting the main loop
2. teardown, fires during cleanup

Alternately, I could allow passing a Service object, in addition to the current dict spec, which would let you define your own start/stop behavior (pretty much equivalent to above). This might have the benefit of being more clearly not a general extension mechanism.